torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -454,7 +454,9 @@ void PropagateShapeOnBlock(Block * block, bool insert_expands) {
 void PropagateInputShapes(Graph & graph, const ArgumentSpec & spec) {
   JIT_ASSERT(graph.inputs().size() == spec.size());
   for(size_t i = 0; i < spec.size(); ++i) {
-    graph.inputs()[i]->setType(spec.tensorInfo(i));
+    auto argspec = spec.at(i);
+    if (argspec.kind() != IValueKind::Tensor) continue;
```

Comment:
This seems to be the only use of .kind. It could be replaced with 

1. having `argspec.isTensor()`, or
2. `if(graph.inputs[i]->isSubtypeOf(DynamicType::get()))`

Reply:
Good point, will do that.